photon-api/src/integTest/scala/com/linkedin/photon/ml/projector/IndexMapProjectorRDDTest.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.projector
+
+import org.testng.Assert._
+import org.testng.annotations.Test
+
+import com.linkedin.photon.ml.test.SparkTestUtils
+import com.linkedin.photon.ml.util.GameTestUtils
+
+/**
+ * Integration tests for [[IndexMapProjectorRDD]].
+ */
+class IndexMapProjectorRDDTest extends SparkTestUtils with GameTestUtils {
+
+  @Test
+  def testBuildIndexMapProjector(): Unit = sparkTest("testBuildIndexMapProjector") {
+    val dataSet1 = generateRandomEffectDataSet(
+      randomEffectIds = Seq("1"),
+      randomEffectType = "per-item",
+      featureShardId = "itemShard",
+      size = 20,
+      dimensions = 7)
+
+    val dataSet2 = generateRandomEffectDataSet(
+      randomEffectIds = Seq("1"),
+      randomEffectType = "per-item",
+      featureShardId = "itemShard",
+      size = 20,
+      dimensions = 10)
+
+    val projector = IndexMapProjectorRDD.buildIndexMapProjector(dataSet1)
+    val projected = projector.projectRandomEffectDataSet(dataSet2)
+
+    val projectedDimentions = projected
+      .activeData
+      .map { case (_, localDataSet) => localDataSet.dataPoints.head }
+      .map { case (_, labeledPoint) => labeledPoint.features.length }
+      .take(1)(0)
+
+    assertEquals(projectedDimentions, 6)
```

Comment:
Why is the result `6`? Shouldn't it be `7`, since that is the dimension the data for dataSet1?

Reply:
Seems to be a quirk of the random vector generator. I'll double check though.

Reply:
It turned out that `generateRandomEffectDataSet` generates a lot of feature values near zero, so they were being filtered out. I added a new function that allows specific exact feature vectors for the purposes of this test.